pdc/apps/package/serializers.py

```diff
@@ -221,3 +222,79 @@ class BuildImageRTTTestsSerializer(StrictSerializerMixin, serializers.ModelSeria
+    build = serializers.CharField(required=False, default=None)
+    # base on content.format, currently rpm.srpm_name
+    package = serializers.CharField(required=False, default=None)
+    # base on content.format, currently rpm.filename
+    file = serializers.CharField(required=False, default=None)
+    # base on content.format, pk
+    file_primary_key = serializers.IntegerField(allow_null=False)
+    repo = RepositoryField()
+    released_date = serializers.DateField(required=False)
+    release_date = serializers.DateField()
+    created_at = serializers.DateTimeField(required=False, read_only=True)
+    updated_at = serializers.DateTimeField(required=False, read_only=True)
+    zero_day_release = serializers.BooleanField(required=False, default=False)
+    obsolete = serializers.BooleanField(required=False, default=False)
+
+    class Meta:
+        model = models.ReleasedFiles
+        fields = ('id', 'build', 'package', 'file', 'file_primary_key', 'repo',
+                  'released_date', 'release_date', 'created_at', 'updated_at',
+                  'zero_day_release', 'obsolete')
+
+    def validate(self, data):
+        if "repo" in data:
+            repo_format = data["repo"].content_format
+            repo_name = data["repo"].name
+            if str(repo_format) != "rpm":
+                raise serializers.ValidationError(
+                    {'detail': 'Currently we just support rpm type of repo, the type of %s is %s ' % (repo_name, repo_format)})
+
+        if "file_primary_key" in data:
+            d = models.RPM.objects.get(id=data["file_primary_key"])
+            if "build" in data:
+                if data["build"]:
+                    if "%s-%s-%s" % (d.srpm_name, d.version, d.arch) != data["build"]:
+                        raise serializers.ValidationError(
+                            {'detail': 'Build should be %s' % ("%s-%s-%s" % (d.srpm_name, d.version, d.arch))})
+                    else:
+                        data["build"] = "%s-%s-%s" % (d.srpm_name, d.version, d.arch)
+            else:
+                data["build"] = "%s-%s-%s" % (d.srpm_name, d.version, d.arch)
+
+            if "package" in data:
+                if data["package"]:
+                    if d.srpm_name != data["package"]:
+                        raise serializers.ValidationError(
+                            {'detail': 'Package should be %s' % d.srpm_name})
+                else:
+                    data["package"] = d.srpm_name
+            else:
+                data["package"] = d.srpm_name
+
+            if "file" in data:
+                if data["file"]:
+                    if d.filename != data["file"]:
+                        raise serializers.ValidationError(
+                            {'detail': 'File should be %s' % d.filename})
+                else:
+                    data["file"] = d.filename
+            else:
+                data["file"] = d.filename
```

Comment:
So "build", "package" and "file" can contain only specific string depending on RPM data?

If I modify data in rpms, these values here can become incorrect.

Looks like these values don't need to be stored at all since they're available in RPM.

Reply:
Good point, seems there is no way to let it know things have changed.
Will remove the three column there, and give comment on UI to let ct get the three files by file_primary_key.

@simozhan  What's your thinking of this part?

Reply:
I have remove the three columns in db, but list them when ct get data, we can get it from rpms by pk. New patch commit.